client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-domains/use-price-override.ts

```diff
@@ -0,0 +1,58 @@
+import { doesStringResembleDomain } from '@automattic/onboarding';
+import { useI18n } from '@wordpress/react-i18n';
+import { useDebounce } from 'use-debounce';
+import { useGetDomainPriceOverride } from 'calypso/landing/stepper/hooks/use-get-domain-price-override';
+
+type Vendor = '100-year-domains';
+export type DomainPriceOverrideOptions = {
+	vendor?: Vendor;
+};
+
+export function usePriceOverride( domain: string, options: DomainPriceOverrideOptions = {} ) {
```

Comment:
The PR also uses a child-custom hook to rely on the data returned by the `useQuery` one, so we don't expose its data externally, and the state is handled within it rather than within its consumers.